src/scripts/librarymenu.js

```diff
@@ -155,297 +302,671 @@ define(["dom", "layoutManager", "inputManager", "connectionManager", "events", "
+        var apiClient = getCurrentApiClient();
+        var libraryMenuOptions = document.querySelector(".libraryMenuOptions");
+
+        if (libraryMenuOptions) {
+            getUserViews(apiClient, userId).then(function (result) {
+                var items = result;
+                var html = "";
+                html += '<h3 class="sidebarHeader">';
+                html += globalize.translate("HeaderMedia");
+                html += "</h3>";
+                html += items.map(function (i) {
+                    var icon = "folder";
+                    var itemId = i.Id;
+
+                    if ("channels" == i.CollectionType) {
+                        itemId = "channels";
+                    }
+                    else if ("livetv" == i.CollectionType) {
+                        itemId = "livetv";
+                    }
+
+                    else if ("photos" == i.CollectionType) {
+                        icon = "photo_library";
+                    }
+                    else if ("music" == i.CollectionType || "musicvideos" == i.CollectionType) {
+                        icon = "library_music";
+                    }
+                    else if ("books" == i.CollectionType) {
+                        icon = "library_books";
+                    }
+                    else if ("playlists" == i.CollectionType) {
+                        icon = "view_list";
+                    }
+                    else if ("games" == i.CollectionType) {
+                        icon = "games";
+                    }
+                    else if ("movies" == i.CollectionType) {
+                        icon = "video_library";
+                    }
+                    else if ("channels" == i.CollectionType || "Channel" == i.Type) {
+                        icon = "videocam";
+                    }
+                    else if ("tvshows" == i.CollectionType) {
+                        icon = "tv";
+                    }
+                    else if ("livetv" == i.CollectionType) {
+                        icon = "live_tv";
+                    }
+                    icon = i.icon || icon;
+
+                    if (i.onclick) {
+                        i.onclick;
+                    }
+
+                    return '<a is="emby-linkbutton" data-itemid="' + itemId + '" class="lnkMediaFolder navMenuOption" href="' + getItemHref(i, i.CollectionType) + '"><i class="md-icon navMenuOptionIcon">' + icon + '</i><span class="sectionName navMenuOptionText">' + i.Name + "</span></a>";
+                }).join("");
+                libraryMenuOptions.innerHTML = html;
+                var elem = libraryMenuOptions;
+                var sidebarLinks = elem.querySelectorAll(".navMenuOption");
+
+                for (var i = 0, length = sidebarLinks.length; i < length; i++) {
+                    sidebarLinks[i].removeEventListener("click", onSidebarLinkClick);
+                    sidebarLinks[i].addEventListener("click", onSidebarLinkClick);
```

Comment:
What's the point of this?

Reply:
Fuck if I know. This is just the deuglify with my single fix. Didn't delve too deep into what unrelated parts of code did.